tests/Server/Workspace/DidChangeWatchedFilesTest.php

```diff
@@ -0,0 +1,43 @@
+<?php
+declare(strict_types = 1);
+
+namespace LanguageServer\Tests\Server\Workspace;
+
+use LanguageServer\ContentRetriever\FileSystemContentRetriever;
+use LanguageServer\{DefinitionResolver, LanguageClient, PhpDocumentLoader, Server};
+use LanguageServer\Index\{DependenciesIndex, Index, ProjectIndex};
+use LanguageServer\Protocol\{FileChangeType, FileEvent, Message};
+use LanguageServer\Tests\MockProtocolStream;
+use LanguageServer\Tests\Server\ServerTestCase;
+use LanguageServer\Server\Workspace;
+use Sabre\Event\Loop;
+
+class DidChangeWatchedFilesTest extends ServerTestCase
+{
+    public function testDeletingFileClearsAllDiagnostics()
+    {
+        $client = new LanguageClient(new MockProtocolStream(), $writer = new MockProtocolStream());
+        $projectIndex = new ProjectIndex($sourceIndex = new Index(), $dependenciesIndex = new DependenciesIndex());
+        $definitionResolver = new DefinitionResolver($projectIndex);
+        $loader = new PhpDocumentLoader(new FileSystemContentRetriever(), $projectIndex, $definitionResolver);
+        $workspace = new Server\Workspace($client, $projectIndex, $dependenciesIndex, $sourceIndex, null, $loader, null);
+
+        $fileEvent = new FileEvent();
+        $fileEvent->uri = 'my uri';
+        $fileEvent->type = FileChangeType::DELETED;
```

Comment:
I think it makes sense to add a constructor to `FileEvent` to make this easier to construct